app/javascript/components/SignUp.jsx

```diff
@@ -29,29 +53,78 @@ class SignUp extends Component {
         this.formSubmit(e.target);
     };
 
-    async formSubmit(formData){
-        var data = new FormData(formData);
+    async formSubmit(){
         await fetch('/signup_user', {
-            Accept: 'application/json',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
             method: "POST",
-            mode: "cors",
-            body: data
-        }).then(response => response.text())
-        .then(res => console.log(res))
+            body: JSON.stringify({
+                name: this.state.name,
+                phone: this.state.phone,
+                email: this.state.email,
+                birthdate: `${this.state.day} ${this.state.month} ${this.state.year}`,
+                password_digest: this.state.password_digest,
+                username: this.state.username, 
+                private: this.state.private,
+                profile_picture: this.state.profile_picture
+            })
+        }).then(response => {
+            response.json();
+            this.props.history.push({
+                pathname: "/home",
+                state: {cur_usr_username: this.state.username}
+            });
+        });
+    }
+
+    onChangeBdayOpt(e){
```

Comment:
It's better to be more verbose. We could lose the meaning eventually and it will be hard to know what this function was intended for. Let's change it to `onChangeBirthdateOption`.